antarest/launcher/adapters/factory_launcher.py

```diff
@@ -32,14 +32,16 @@ def build_launcher(
         cache: ICache,
     ) -> Dict[str, AbstractLauncher]:
         dict_launchers: Dict[str, AbstractLauncher] = dict()
-        if config.launcher.local is not None:
-            dict_launchers["local"] = LocalLauncher(config, callbacks, event_bus, cache)
-        if config.launcher.slurm is not None:
-            dict_launchers["slurm"] = SlurmLauncher(
-                config,
-                callbacks,
-                event_bus,
-                cache,
-                retrieve_existing_jobs=True,
-            )
+        for cfg in config.launcher.configs or []:
+            if isinstance(cfg, SlurmConfig):
```

Comment:
In relation to the previous comment: this is indeed the only legitimate place to do the dispatch slurm/local, all other specificities should be handled by the launchers (by the implementation class) that we instantiate here.